### ISSUE TITLE
[a11y]: Investigate reading/focus order of Tearsheet elements

### ISSUE THREAD

**Issue by elycheea** [carbon-design-system/ibm-products#7851]
### Package

@carbon/ibm-products

### Package version

v2.70.1

### Automated testing tool and ruleset

manual

### Assistive technology

JAWS

### Description

Reported by Raghu during screen reader testing.

Reading order is not following the visual order.
The close the tearsheet button is visually presented in the top right part of the tearsheet, however, on <kbd>Arrow</kbd> key navigation this element announced at the middle of the modal (just after tab list)

**Recommended fix**
Make this element identified in the top of the modal, where it is visually presented. (On Both <kbd>Tab</kbd> / <kbd>Arrow</kbd> key navigation.)

**User impact**
When tab focus is not in a meaningful sequence, it confuses users who are either blind or have motor and cognitive disabilities.

> [!NOTE]
> This may be specific to our Storybook implementations, but validated that there is some confusion in the currently ordering of elements in the “tearsheet with all header items and influencer” story.
>
> We should investigate ordering of AI label and close button. These appear to focus out of order, but the `ComposedModal` focuses these in the correct order.
>
> We should see if we can improve this ordering. It may be more specific to highlighting the AI label by default that results in ordering feeling off.

### WCAG 2.1 Violation

2.4.3. Focus order

### Reproduction/example

https://ibm-products.carbondesignsystem.com/?path=/story/components-tearsheet--fully-loaded

### Steps to reproduce

1. Navigate to the [tearsheet with all header items and influencer story](https://ibm-products.carbondesignsystem.com/?path=/story/components-tearsheet--fully-loaded).
  - The AI label should be focused by default.
2. Continue to tab to the next interactive items.
  - The header actions then tab list are focused before the close button is focused, which is visually next to the AI label.
3. Compare against the [composed modal with AI label story](https://react.carbondesignsystem.com/?path=/story/components-composedmodal--with-ai-label).
  - Form items are focused before navigating back to AI label and then close sequentially.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems

**Comment by tay1orjones** [comment]
Let me know how this resonates https://github.com/carbon-design-system/carbon/issues/20626#issuecomment-3357779895

**Comment by amal-k-joy** [comment]
> Let me know how this resonates [carbon-design-system/carbon#20626 (comment)](https://github.com/carbon-design-system/carbon/issues/20626#issuecomment-3357779895)

@tay1orjones ,
I think this issue would be there still,
since close button would be focussed only after tabs, since tabs are kept as header content.

cc: @matthewgallo

**Comment by amal-k-joy** [comment]
We’ve adopted a solution in the new Tearsheet to address this. We hide the ComposeModal close button and AI label, and instead render them within the Tearsheet. This gives us full control over the DOM order.
